bls/src/keypair.rs

```diff
@@ -157,6 +190,28 @@ impl TryFrom<&Pubkey> for PubkeyProjective {
     }
 }
 
+impl TryFrom<&[u8]> for PubkeyProjective {
+    type Error = BlsError;
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        if bytes.len() != BLS_PUBLIC_KEY_AFFINE_SIZE {
+            return Err(BlsError::ParseFromBytes);
+        }
+        // unwrap safe due to the length check above
+        let public_affine = Pubkey(bytes.try_into().unwrap());
+
+        public_affine.try_into()
+    }
+}
+
+impl From<&PubkeyProjective> for [u8; BLS_PUBLIC_KEY_AFFINE_SIZE] {
+    fn from(pubkey: &PubkeyProjective) -> Self {
+        let mut bytes = [0u8; BLS_PUBLIC_KEY_AFFINE_SIZE];
+        let pubkey_affine: Pubkey = (*pubkey).into();
+        bytes.copy_from_slice(pubkey_affine.0.as_slice());
+        bytes
```

Comment:
Same with this one, since the inner type is public, you can just do
```suggestion
        let pubkey_affine: Pubkey = (*pubkey).into();
        pubkey_affine.0
```